.gitlab-ci.yml

```diff
@@ -286,147 +206,149 @@ build-core:
     - docker
   rules:
     - changes:
-        - dockerfiles/**/*
+        - docker/**/*
         - scripts/**/*
-        - common/**/*
         - core/**/*
+        - common/**/*
         - CMakeLists.txt
         - cmake/**/*
         - .gitlab-ci.yml
       when: on_success
-    - exists:
-      - "build-core-base"
-      when: on_success
   script:
     - BRANCH_LOWER=$(echo "$CI_COMMIT_REF_NAME" | tr '[:upper:]' '[:lower:]')
     - echo "$BRANCH_LOWER"
-    - env
     - ./scripts/generate_datafed.sh
-    - cat ./config/datafed.sh
     - docker login "${REGISTRY}" -u "${HARBOR_USER}" -p "${HARBOR_DATAFED_GITLAB_CI_REGISTRY}"
-    - docker build --build-arg BASE_CORE_IMAGE="${REGISTRY}/${IMAGE_TAG}-base-${BRANCH_LOWER}:latest" -f core/docker/Dockerfile.core.ubuntu -t "${REGISTRY}/${IMAGE_TAG}-${BRANCH_LOWER}" .
+    - docker build --build-arg DEPENDENCIES="${REGISTRY}/datafed/dependencies-${BRANCH_LOWER}:latest" --build-arg RUNTIME="${REGISTRY}/datafed/runtime-${BRANCH_LOWER}:latest" -f core/docker/Dockerfile -t "${REGISTRY}/${IMAGE_TAG}-${BRANCH_LOWER}:latest" .
```

Comment:
@nedvedba I want you to point things out like this, there is a lot of repetition between the different docker builds, and also in how the containers are run.  From an operations point of view it makes sense to have those items in a operational type scripts so someone can just pick the script up and run it.